### PR TITLE
Add missing NSPrivacyCollectedDataTypes to PrivacyInfo.xcprivacy

### PR DESCRIPTION
**Pull request by DandyLyons**
Revision of: https://github.com/pointfreeco/swift-composable-architecture/pull/2930
Per: https://github.com/pointfreeco/swift-composable-architecture/issues/3021

The Xcode behavior appears to be different from when I submitted PR 2930. I originally created the manifest without NSPrivacyCollectedDataTypes (since TCA is not actually collecting anything) and Xcode generated an empty Privacy Report. 

Now it generates the error `Missing an expected key: 'NSPrivacyCollectedDataTypes'`

I've now added the key in however, it no longer generates an empty Privacy Report, so I think it needs to be a library decision how we proceed. Here's the new generated report. 

Screenshot 2024-04-29 at 1 10 41 PM

According to this we are collecting "other" data even though TCA isn't collecting anything and even though we already have an Apple-approved exemption (C56D.1). I believe that this is misleading to library users, and Apple may force app makers to include this in their final App Store nutrition label declaration, which would be further misleading to customers.

### CODE REVIEW

**Comment by DandyLyons** [comment]
It's not an array. It's a string. 

Also the NSPrivacyCollectedDataType is not allowed to be an empty string. And it must be one of the approved strings. 
[TransformFeature-PrivacyReport 2024-04-29 12-06-51.pdf](https://github.com/pointfreeco/swift-composable-architecture/files/15153922/TransformFeature-PrivacyReport.2024-04-29.12-06-51.pdf)

File: Sources/ComposableArchitecture/Resources/PrivacyInfo.xcprivacy [hunk 1]
```diff
@@ -2,6 +2,21 @@
...
+			<key>NSPrivacyCollectedDataTypeLinked</key>
+			<false/>
+			<key>NSPrivacyCollectedDataTypeTracking</key>
+			<false/>
+			<key>NSPrivacyCollectedDataTypePurposes</key>
+			<array>
+				<string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
+			</array>
+		</dict>
+	</array>
```
Comment (stephencelis) [inline Sources/ComposableArchitecture/Resources/PrivacyInfo.xcprivacy:19]: @DandyLyons This is the array I was referring to:

```suggestion
	<array/>
```

Does it not work to pass in an empty array?

**Comment by DandyLyons** [comment]
Looks good with the empty array now. It generates an empty Privacy Report again.

**Review by stephencelis — APPROVED** [review]
Thanks!

**Pull request merged**